### PR TITLE
Content page

### PR DESCRIPTION
**Pull request by mattgrill**
A content page.

- [x] Filters
- [x] Sorting.

### CODE REVIEW

**Comment by mattgrill** [comment]
Alright, 👍 from me. I'll merge this now and when we get to the point of having some custom styles for the material-ui components we can start the wrapping of the components with our namespace/styles.

File: src/actions/application.js [hunk 1]
```diff
@@ -48,6 +48,37 @@ function* loadMenu() {
   }
 }
 
+/**
+ * Gets all available content types.
+ */
+export const CONTENT_TYPES_REQUESTED = 'CONTENT_TYPES_REQUESTED';
```
Comment (dawehner) [inline src/actions/application.js]: It feels really weird to put these reducers into application.js. Is there a reason this is not moved to a admin/content specific page?

File: src/utils/api/api.js [hunk 2]
```diff
@@ -36,6 +36,14 @@ async function api(
     case 'permissions':
       url = '/admin-api/permissions?_format=json';
       break;
+    case 'content':
```
Comment (dawehner) [inline src/utils/api/api.js]: It is weird that there is 'content' and 'nodeType' :)

File: src/components/05_pages/Content/Content.js [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+            <MenuItem value="">
+              <em>Any</em>
+            </MenuItem>
+            <MenuItem value="published">Published</MenuItem>
+            <MenuItem value="unpublished">Unpublished</MenuItem>
+          </Select>
+        </FormControl>
+      </div>
+      <Paper>
+        <Table>
```
Comment (lauriii) [inline src/components/05_pages/Content/Content.js]: Should we show a loading animation at least on the initial page load?

File: src/components/05_pages/Content/Content.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+              ].map(item => <TableCell key={item}>{item}</TableCell>)}
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {this.props.nodes.map(
+              ({ type, attributes: { changed, nid, status, title } }) => (
+                <TableRow key={nid}>
+                  <TableCell>{title}</TableCell>
+                  <TableCell>{this.props.contentTypes[type].name}</TableCell>
+                  <TableCell />
```
Comment (lauriii) [inline src/components/05_pages/Content/Content.js]: Should we remove this column for now and create a follow-up issue to add it?

File: src/components/05_pages/Content/Content.js [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+                    </Button>
+                    <Button
+                      variant="fab"
+                      disabled
+                      aria-label="delete"
+                      className={styles.button}
+                      component={Link}
+                      to={`/node/${nid}/delete`}
+                    >
+                      <DeleteIcon />
```
Comment (lauriii) [inline src/components/05_pages/Content/Content.js]: Should we link this to the delete page for now?

File: src/components/05_pages/Content/Content.js (hunk 3 above)
Comment (justafish) [reply src/components/05_pages/Content/Content.js]: @lauriii I removed the LoadingBar from this page as I think we should move it to the Default template, but didn't want to start exploring that here

Comment (lauriii) [reply src/components/05_pages/Content/Content.js]: @justafish makes sense! Let's not block this PR with that, but we should open an issue for that anyway! 🙂

File: src/components/05_pages/Content/Content.js (hunk 4 above)
Comment (mattgrill) [reply src/components/05_pages/Content/Content.js]: This is the user?

File: src/actions/application.js (hunk 1 above)
Comment (justafish) [reply src/actions/application.js]: @dawehner exactly :+1:

File: src/components/05_pages/Content/Content.js [hunk 6]
```diff
@@ -157,7 +157,6 @@ class Content extends Component {
               {[
                 'Title',
                 'Content Type',
-                'Author',
```
Comment (mattgrill) [inline src/components/05_pages/Content/Content.js]: I'll take care of adding the author in a followup. It shouldn't be too difficult.

File: src/App.js [hunk 7]
```diff
@@ -35,29 +37,40 @@ const store = createStore(
...
+  <JssProvider jss={jss} generateClassName={generateClassName}>
+    <ErrorBoundary>
+      <Provider store={store}>
+        <ConnectedRouter history={history}>
+          <Switch>
             <Route
-              path={route}
-              component={withDefault(withRouter(routes[route]))}
-              key={route}
+              exact
```
Comment (dawehner) [inline src/App.js]: In order to replicate the behaviour of Drupal right now we would need to check "access content overview" on this route. Of course we can do this later, but I think access checking on routes, not just endpoints, is something we should keep in mind somehow.

**Pull request merged**